### PR TITLE
Enable naive Pipeline Parallelism training for Gpt neox japanese and san japanese

### PR DESCRIPTION
**Pull request by mayankagarwals**
As suggested in the https://github.com/huggingface/transformers/issues/22561, moved labels to the same device as logits for GPT Neox Japanese and Gpt San Japanese

### CODE REVIEW

**Comment by mayankagarwals** [comment]
@sgugger Could you review this once?

**Review by sgugger — APPROVED** [review]
Thanks a lot!
